core/server/services/email-analytics/lib/event-processor.js

```diff
@@ -88,6 +88,17 @@ class GhostEventProcessor extends EventProcessor {
                 opened_at: this.db.knex.raw('COALESCE(opened_at, ?)', [moment.utc(event.timestamp).format('YYYY-MM-DD HH:mm:ss')])
             });
 
+        const {value: timezone} = await this.db.knex('settings').first('value').where('key', 'timezone');
+        await this.db.knex('members')
+            .where('email', '=', event.recipientEmail)
+            .andWhere(builder => builder
+                .where('last_seen_at', '<', moment.utc(event.timestamp).tz(timezone).startOf('day').format('YYYY-MM-DD HH:mm:ss'))
```

Comment:
There's an error here, because the `format()` is called on the date in the site's timezone so will always be `00:00:00`. It needs another call to `utc()` after the `startOf()` to compare against the UTC date in the db:

```
> moment.utc().tz('Europe/Paris').startOf('day').format('YYYY-MM-DD HH:mm:ss')
'2022-03-01 00:00:00'
> moment.utc().tz('Europe/Paris').startOf('day').utc().format('YYYY-MM-DD HH:mm:ss')
'2022-02-28 23:00:00'
```

Reply:
I tried to make a failing test on https://github.com/TryGhost/Members/blob/main/packages/members-events-service/lib/last-seen-at-updater.js but it couldn't fail. This is a bug specific to the SQL implementation so I'm just adding this missing `.utc()` and we're good